ui/src/containers/ConnectionLostDialog.jsx

```diff
@@ -1,88 +1,38 @@
-/* eslint-disable react/jsx-handler-names */
-/* eslint-disable react/no-string-refs */
-import React from 'react';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
-import { Modal, Button } from 'react-bootstrap';
-import { showConnectionLostDialog } from '../actions/general';
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { Modal, Button, Spinner } from 'react-bootstrap';
 
-export class ConnectionLostDialog extends React.Component {
-  constructor(props) {
-    super(props);
-    this.accept = this.accept.bind(this);
-    this.reject = this.reject.bind(this);
-    this.intervalCb = this.intervalCb.bind(this);
-    this.seconds = 60;
-    this.timerid = undefined;
-  }
-
-  intervalCb() {
-    this.seconds--;
-
-    if (this.refs.countdown) {
-      this.refs.countdown.innerHTML = this.seconds;
-    }
-
-    if (this.seconds < 1) {
-      this.seconds = 0;
-      this.accept();
-    }
-  }
-
-  accept() {
-    this.forceUpdate();
-  }
+export function ConnectionLostDialog() {
+  const [buttonLoading, setButtonLoading] = useState(false);
```

Comment:
I wouldn't even keep this state. If users click "Reload", we want to reload the page as quickly as possible.